lib/ansible/modules/windows/win_disk_facts.py

```diff
@@ -0,0 +1,402 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2017, Marc Tschapek <marc.tschapek@itelligence.de>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = r'''
+---
+module: win_disk_facts
+version_added: '2.5'
+short_description: Show the attached disks and disk information of the target host
+description:
+   - With the module you can retrieve and output detailed information about the attached disks of the target and
+     it's volumes and partitions if existent.
+requirements:
+    - Windows 8.1 / Windows 2012R2 (NT 6.3) or newer
+author:
+    - Marc Tschapek (@marqelme)
+notes:
+  - You can use this module in combination with the win_disk_management module in order to retrieve the disks status
+    on the target.
+  - In order to understand all the returned properties and values please visit the following site and open the respective MSFT class
+    https://msdn.microsoft.com/en-us/library/windows/desktop/hh830612(v=vs.85).aspx
+'''
+
+EXAMPLES = r'''
+- name: get disk facts
+  win_disk_facts:
+- name: output first disk size
+  debug:
+    var: ansible_facts.ansible_disk.disk_0.size
+
+- name: get disk facts
+  win_disk_facts:
+- name: output second disk serial number
+  debug:
+    var: ansible_facts.ansible_disk.disk_1.serial_number
+'''
+
+RETURN = r'''
+changed:
```

Comment:
You don't need to document `changed` and `msg` as they are generic return values.

Reply:
removed